tasks/main.yml

```diff
@@ -3,3 +3,4 @@
     command: "{{ item.command }}"
     value: "{{ item.value }}"
   with_items: "{{ tasmota_commands }}"
+  when: item.value is defined and item.value
```

Comment:
Without tested it I would assume that this will not work in all cases. Perhaps if you set the value of `0`.
It is also not best practice to use a value without casting it to `bool`.

I suggest to use the `omit` construct here. Have a look here: https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#making-variables-optional

Sadly I can not give you a hint how to test this in the action_plugin but I assume that there is some helper from Ansible side you can use

Reply:
Yeah, i also tried to use `omit` but this is some weird behavior. It basically sets the value to `__omit_place_holder__...` and I was not able to find a build-in conditional that matches here. So we'd need to use something like `startwith("__omit_place_holder")`.

I really missed the `0`, mainly cause I use `"0"`. What do you think about:
`when: item.value | string | default(False)` 
?

This works for both `0` and `"0"`, booleans and foo.

Reply:
I did some research and I think this is handled internally by ansible. When setting the value to `omit` Ansible should not hand over the value to the action_plugin so the `value` should not be defined. Maybe we can check if `value` is defined and mark the step as skipped when it is not?

But I am also fine with the current solution when my suggested solution is not possible.

Reply:
Ansible says no:

```
failed: [tasmota_katzenbrunnen] (item={u'command': u'Gpio17', u'value': u'__omit_place_holder__4a90a841447c3a40814a98c725d718b3f85596e5'}) => {
    "ansible_loop_var": "item", 
    "changed": false, 
    "item": {
        "command": "Gpio17", 
        "value": "__omit_place_holder__4a90a841447c3a40814a98c725d718b3f85596e5"
    }, 
    "msg": "error during retrieving parameter 'parameter value is required'"
}
```

Reply:
@deveth0 acutally it was exactly working as i expected. Not Ansible was complaining about the missing parameter. The Action plugin expected the parameter to be set.

Please try https://github.com/tobias-richter/ansible-tasmota/pull/11 which should work with the omit placeholder.

Reply:
oh yeah, that works fine!
```
    - command: Gpio1
      value: "{{gpio1 | default(omit)}}"
```
=> 
```
skipping: [tasmota_futterautomat] => (item={u'command': u'Gpio1', u'value': u'__omit_place_holder__3cf3a2d689ce9fc257beb65030a14acf0c5e2386'})
```